R/discovery_prediction.R

```diff
@@ -86,31 +104,42 @@ discover_signatures <- function(musica, table_name, num_signatures,
   return(result)
 }
 
-#' LDA prediction of samples based on existing signatures
-#'
+#' @title Prediction of exposures in new samples using pre-existing signatures
+#' @description Exposures for samples will be predicted using an existing set
+#' of signatures stored in a \code{\linkS4class{musica_result}} object. 
+#' Algorithms available for prediction include a modify version of \code{"lda"},
+#' \code{"decompTumor2Sig"}, and \code{"deconstructSigs"}.
 #' @param musica A \code{\linkS4class{musica}} object.
 #' @param g A \linkS4class{BSgenome} object indicating which genome
-#' reference the variants and their coordinates were derived from.
+#' reference the variants and their coordinates were derived from. Only used
+#' if \code{algorithm = "deconstructSigs"}
 #' @param table_name Name of table used for posterior prediction.
 #' Must match the table type used to generate the prediction signatures
-#' @param signature_res Signatures to use for prediction
-#' @param algorithm Algorithm to use for prediction. Choose from
-#' "lda_posterior", decompTumor2Sig, and deconstructSigs
-#' @param signatures_to_use Which signatures in set to use (default all)
-#' @param verbose Whether to show intermediate results
-#' @return A result object containing signatures and sample weights
+#' @param signature_res Signatures used to predict exposures for the samples
+#' \code{musica} object. Existing signatures need to stored in a
+#' \code{\linkS4class{musica_result}} object. 
+#' @param algorithm Algorithm to use for prediction of exposures. One of
+#' \code{"lda"}, \code{"decompTumor2Sig"}, or
+#' \code{"deconstructSigs"}.
+#' @param signatures_to_use Which signatures in the \code{signature_res} result
+#' object to use. Default is to use all signatures.
+#' @param verbose If \code{TRUE}, progress will be printing. Only used if
+#' \code{algorithm = "lda"}. Default \code{FALSE}.
+#' @return Returns a A \code{\linkS4class{musica_result}} object containing
+#' signatures given by the \code{signature_res} parameter and exposures
+#' predicted from these signatures.
 #' @examples
 #' data(musica)
 #' data(cosmic_v2_sigs)
 #' g <- select_genome("19")
 #' build_standard_table(musica, g, "SBS96", overwrite = TRUE)
-#' predict_exposure(musica = musica, table_name = "SBS96",
+#' result <- predict_exposure(musica = musica, table_name = "SBS96",
 #' signature_res = cosmic_v2_sigs, algorithm = "lda")
 #' 
-#' #Predict using seed
-#' seed <- 1
-#' withr::with_seed(seed, predict_exposure(musica = musica, 
-#' table_name = "SBS96", signature_res = cosmic_v2_sigs, algorithm = "lda"))
+#' # Predict using LDA-like algorithm with seed set to 1
```

Comment:
Should we give two examples, one with set.seed() and one with with_seed()? I think different users may want one or the other.